tests/src/test/java/alluxio/server/ft/journal/raft/FaultToleranceEBJTest.java

```diff
@@ -0,0 +1,278 @@
+      File[] files = snapshotDir.listFiles();
+      return files != null && files.length > 1 && files[0].length() > 0;
+    }, WaitForOptions.defaults().setInterval(RETRY_INTERVAL_MS).setTimeoutMs(RESTART_TIMEOUT_MS));
+  }
+
+  @Test
+  public void restart() throws Exception {
+    standardBefore();
+
+    AlluxioURI testDir = new AlluxioURI("/dir");
+    FileSystem fs = mCluster.getFileSystemClient();
+    fs.createDirectory(testDir);
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    restartMasters();
+    assertTrue(fs.exists(testDir));
+    mCluster.notifySuccess();
+  }
+
+  @Test
+  public void restartStress() throws Throwable {
+    standardBefore();
+
+    // Run and verify operations while restarting the cluster multiple times.
+    AtomicReference<Throwable> failure = new AtomicReference<>();
+    AtomicInteger successes = new AtomicInteger(0);
+    FileSystem fs = mCluster.getFileSystemClient();
+    List<FaultToleranceEBJTest.OperationThread> threads = new ArrayList<>();
+    try {
+      for (int i = 0; i < 10; i++) {
+        FaultToleranceEBJTest.OperationThread t =
+                new FaultToleranceEBJTest.OperationThread(fs, i, failure, successes);
+        t.start();
+        threads.add(t);
+      }
+      for (int i = 0; i < 2; i++) {
+        restartMasters();
+        System.out.printf("---------- Iteration %s ----------\n", i);
+        successes.set(0);
+        CommonUtils.waitFor("11 successes", () -> successes.get() >= 11,
+                WaitForOptions.defaults().setTimeoutMs(RESTART_TIMEOUT_MS));
+        if (failure.get() != null) {
+          throw failure.get();
+        }
+      }
+    } finally {
+      threads.forEach(Thread::interrupt);
+      for (Thread t : threads) {
+        t.join();
+      }
+    }
+    mCluster.notifySuccess();
+  }
+
+  private void restartMasters() throws Exception {
+    for (int i = 0; i < NUM_MASTERS; i++) {
+      mCluster.stopMaster(i);
+    }
+    for (int i = 0; i < NUM_MASTERS; i++) {
+      mCluster.startMaster(i);
+    }
+  }
+
+  private static class OperationThread extends Thread {
+    private final FileSystem mFs;
+    private final int mThreadNum;
+    private final AtomicReference<Throwable> mFailure;
+    private final AtomicInteger mSuccessCounter;
+
+    public OperationThread(FileSystem fs, int threadNum, AtomicReference<Throwable> failure,
+                           AtomicInteger successCounter) {
+      super("operation-test-thread-" + threadNum);
+      mFs = fs;
+      mThreadNum = threadNum;
+      mFailure = failure;
+      mSuccessCounter = successCounter;
+    }
+
+    public void run() {
+      try {
+        runInternal();
+      } catch (Exception e) {
+        e.printStackTrace();
+        mFailure.set(e);
+      }
+    }
+
+    public void runInternal() throws Exception {
+      while (!Thread.interrupted()) {
+        for (int i = 0; i < 100; i++) {
```

Comment:
let's make this a constant and bump it to 1000 now that it runs in parallel with the rest.